src/favorites.ts

```diff
@@ -0,0 +1,51 @@
+import * as vscode from 'vscode';
+
+export class FavoritesProvider
+	implements vscode.TreeDataProvider<FavoritesEntry> {
+	// the events to be able to refresh the views
+	private _onDidChangeTreeData: vscode.EventEmitter<
+		FavoritesEntry | undefined
+	> = new vscode.EventEmitter<FavoritesEntry | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<FavoritesEntry | undefined> = this
+		._onDidChangeTreeData.event;
+	private favorites: FavoritesEntry[] | undefined;
+
+	constructor(globalState: vscode.Memento) {
+		// created from the global state to be kept between sessions
+		this.favorites = globalState.get('favorites');
+	}
+
+	refresh(globalState: vscode.Memento): void {
+		// keeping the history consistent with the global state
+		this.favorites = globalState.get('favorites');
+		this._onDidChangeTreeData.fire();
+	}
+
+	getTreeItem(element: FavoritesEntry): vscode.TreeItem {
+		return element;
+	}
+
+	getChildren(element?: FavoritesEntry): Thenable<FavoritesEntry[]> {
+		if (element) {
+			// our tree has only one level
+			return Promise.resolve([]);
+		} else {
+			// from the root, returns the history list
+			return Promise.resolve(this.favorites ? this.favorites : []);
```

Comment:
Just a suggestion to keep the code lean (the same applies to the history one).

```suggestion
			return Promise.resolve(this.favorites || []);
```

<img src="https://media.giphy.com/media/En7DXs95wY1Ow/giphy.gif" />